drivers/source/serial_io.c

```diff
@@ -65,15 +73,24 @@ obc_error_code_t sciPrintText(unsigned char *text, uint32_t length) {
     /* USER CODE BEGIN */
     // Print text to the serial port using sciSendBytes. Use the mutex to protect the SCI module.
 
-    /* USER CODE END */
+    if (xSemaphoreTake(mutex, UART_MUTEX_BLOCK_TIME) == pdTRUE)
+    {
+        sciSendBytes(text, length);
+        xSemaphoreGive(sciLinMutex);
+        return OBC_ERR_CODE_SUCCESS;
+    }
 }
```

Comment:
This is where the build error comes from. If we fail to take the mutex, you should return an error code.